### PR TITLE
Add prefer_async support via post body

### PR DESCRIPTION
**Pull request by robwoodgate**
Related NUT: https://github.com/cashubtc/nuts/pull/339

# PR Summary

- Add prefer_async to melt requests and wire it through wallet/API/mint so clients can request async processing, with pending response returned immediately.
- Implement mint-side async melt settlement via background task, keeping existing sync flow intact.
- Add CLI pay --prefer-async and a regression test that verifies pending → paid transition
- Note: cashu/mint/ledger.py line endings normalized to LF (diff shows as full rewrite without --ignore-space-change).

## incidental Changes for CI test stability

The first 4 commits in this PR implement the feature, the rest are aimed at stabilizing CI to make this PR green.

- disable DB pooling in tests and dispose wallet DB engines to avoid SQLite locks.
- Hardened CLI tests against option parsing edge cases by separating quote IDs from flags.
- Stabilized regtest and mint test infrastructure to reduce flaky failures: timeouts, DB reset, FakeWallet queue loop binding, early P2PK pubkey validation

### CODE REVIEW

File: cashu/wallet/p2pk.py [hunk 1]
```diff
@@ -54,6 +54,10 @@ async def create_p2pk_lock(
         Returns:
             P2PKSecret: P2PK secret with the given pubkeys, locktime, tags, and signature flag.
         """
+        try:
+            PublicKey(bytes.fromhex(data))
+        except Exception as exc:
+            raise ValueError("Invalid pubkey") from exc
```
Comment (a1denvalu3) [inline cashu/wallet/p2pk.py:60]: unrelated?

File: cashu/mint/ledger.py
Comment (a1denvalu3) [inline cashu/mint/ledger.py:1]: Looks like the clanker replaced the whole file, making diff impossible.

File: tests/conftest.py [hunk 2]
```diff
@@ -96,6 +96,14 @@ async def start_mint_init(ledger: Ledger) -> Ledger:
...
         db = Database("mint", settings.mint_database)
         async with db.connect() as conn:
+            await conn.execute(
+                """
+                SELECT pg_terminate_backend(pid)
+                FROM pg_stat_activity
+                WHERE datname = current_database()
+                AND pid <> pg_backend_pid();
+                """
+            )
```
Comment (a1denvalu3) [inline tests/conftest.py:106]: what does this do?

File: tests/wallet/test_wallet_lightning.py [hunk 3]
```diff
@@ -48,14 +48,15 @@ async def reset_wallet_db(wallet: LightningWallet):
...
-    yield wallet
+async def wallet():
+    wallet = await LightningWallet.with_db(
+        url=SERVER_ENDPOINT,
+        db="test_data/wallet1",
+        name="wallet1",
+    )
+    await wallet.async_init()
+    yield wallet
+    await wallet.db.engine.dispose()
```
Comment (a1denvalu3) [inline tests/wallet/test_wallet_lightning.py:59]: Maybe this deserves a PR on its own?

**Review by a1denvalu3 — COMMENTED** [review]
Thank you for the PR, but `prefer_async` is being worked on in #801 . I'll update that one.

File: cashu/mint/ledger.py
Comment (robwoodgate) [reply cashu/mint/ledger.py:1]: I mentioned the line endings in the PR notes. It was using non-standard line endings, so I normalized it.
Use `--ignore-space-changes` to see the raw change. But if you are already working on this feature in a different PR (sorry, didn't see, and the change in spec is < 1 week old) then no need.

File: tests/conftest.py (hunk 2 above)
Comment (robwoodgate) [reply tests/conftest.py:106]: This fixes a CI abnormal exit where the DB is torn down with active threads.
In fact, every commit in this PR after a96882de78d78250b827687b2f3afb5dc89aab17 is trying to fix up CI, as noted in the PR notes

File: tests/wallet/test_wallet_lightning.py (hunk 3 above)
Comment (robwoodgate) [reply tests/wallet/test_wallet_lightning.py:59]: Please be my guest - use or discard anything as you wish

File: cashu/wallet/p2pk.py (hunk 1 above)
Comment (robwoodgate) [reply cashu/wallet/p2pk.py:60]: Related in the sense that CI was intermittently failing tests, and I wanted to remove some of the flakey-ness and get a green run. It's part of the "Incidental changes" noted in the PR summary

File: cashu/mint/ledger.py
Comment (robwoodgate) [reply cashu/mint/ledger.py:1]: Here you go:

```diff
robw:nutshell [prefer-async-post-body] $ git diff 9128bc7 --ignore-space-change cashu/mint/ledger.py
diff --git a/cashu/mint/ledger.py b/cashu/mint/ledger.py
index b60a1f2..422e837 100644
--- a/cashu/mint/ledger.py
+++ b/cashu/mint/ledger.py
@@ -867,6 +867,7 @@ class Ledger(
         proofs: List[Proof],
         quote: str,
         outputs: Optional[List[BlindedMessage]] = None,
+        prefer_async: Optional[bool] = None,
     ) -> PostMeltQuoteResponse:
         """Invalidates proofs and pays a Lightning invoice.
 
@@ -939,6 +940,69 @@ class Ledger(
         if outputs:
             await self._store_blinded_messages(outputs, melt_id=melt_quote.quote)
 
+        if prefer_async:
+            pending_response = PostMeltQuoteResponse.from_melt_quote(
+                melt_quote.model_copy()
+            )
+            asyncio.create_task(
+                self._melt_background_task(
+                    melt_quote=melt_quote,
+                    proofs=proofs,
+                    outputs=outputs,
+                    previous_state=previous_state,
+                    fee_reserve_provided=fee_reserve_provided,
+                    unit=unit,
+                    method=method,
+                )
+            )
+            return pending_response
+        return await self._settle_melt_after_pending(
+            melt_quote=melt_quote,
+            proofs=proofs,
+            outputs=outputs,
+            previous_state=previous_state,
+            fee_reserve_provided=fee_reserve_provided,
+            unit=unit,
+            method=method,
+        )
+
+    async def _melt_background_task(
+        self,
+        *,
+        melt_quote: MeltQuote,
+        proofs: List[Proof],
+        outputs: Optional[List[BlindedMessage]],
+        previous_state: MeltQuoteState,
+        fee_reserve_provided: int,
+        unit: Unit,
+        method: Method,
+    ) -> None:
+        try:
+            await self._settle_melt_after_pending(
+                melt_quote=melt_quote,
+                proofs=proofs,
+                outputs=outputs,
+                previous_state=previous_state,
+                fee_reserve_provided=fee_reserve_provided,
+                unit=unit,
+                method=method,
+            )
+        except Exception as e:
+            logger.error(
+                f"Async melt task failed for quote {melt_quote.quote}: {e}"
+            )
+
+    async def _settle_melt_after_pending(
+        self,
+        *,
+        melt_quote: MeltQuote,
+        proofs: List[Proof],
+        outputs: Optional[List[BlindedMessage]],
+        previous_state: MeltQuoteState,
+        fee_reserve_provided: int,
+        unit: Unit,
+        method: Method,
+    ) -> PostMeltQuoteResponse:
         # if the melt corresponds to an internal mint, mark both as paid
         melt_quote = await self.melt_mint_settle_internally(melt_quote, proofs)
         # quote not paid yet (not internal), pay it with the backend
(END)

```
